test/e2e/baseline_test.go

```diff
@@ -138,12 +140,228 @@ spec:
 	})
 }
 
+func TestServerSideLB(t *testing.T) {
+	framework.NewTest(t).Run(func(t framework.TestContext) {
+		runTestToServiceWaypoint(t, func(t framework.TestContext, src echo.Instance, dst echo.Instance, opt echo.CallOptions) {
+			// Need HTTP
+			if opt.Scheme != scheme.HTTP {
+				return
+			}
+			var singleHost echo.Checker = func(result echo.CallResult, _ error) error {
+				hostnames := make([]string, len(result.Responses))
+				for i, r := range result.Responses {
+					hostnames[i] = r.Hostname
+				}
+				unique := sets.SortedList(sets.New(hostnames...))
+				if len(unique) != 1 {
+					return fmt.Errorf("excepted only one destination, got: %v", unique)
+				}
+				return nil
+			}
+			var multipleHost echo.Checker = func(result echo.CallResult, _ error) error {
+				hostnames := make([]string, len(result.Responses))
+				for i, r := range result.Responses {
+					hostnames[i] = r.Hostname
+				}
+				unique := sets.SortedList(sets.New(hostnames...))
+				want := dst.WorkloadsOrFail(t)
+				wn := []string{}
+				for _, w := range want {
+					wn = append(wn, w.PodName())
+				}
+				if len(unique) != len(wn) {
+					return fmt.Errorf("excepted all destinations (%v), got: %v", wn, unique)
+				}
+				return nil
+			}
+
+			shouldBalance := dst.Config().HasServiceAddressedWaypointProxy()
+			// Istio client will not reuse connections for HTTP/1.1
+			opt.HTTP.HTTP2 = true
+			// Make sure we make multiple calls
+			opt.Count = 10
+			c := singleHost
+			if shouldBalance {
+				c = multipleHost
```

Comment:
do we have workload level waypoint?

Reply:
Not yet, relevant test cases will be added later